app/src/main/java/org/gdg/frisbee/android/appwidget/UpcomingEventWidgetProvider.java

```diff
@@ -124,6 +124,29 @@ public void onNotFound(String key) {
 
         }
 
+        @Nullable
+        private Event getNextEvent(List<Event> listEvents) {
+            if (listEvents == null || listEvents.size() == 0) {
+                return null;
+            }
+            Event ret = listEvents.get(0);
+            for (Event e : listEvents) {
+                if (e.getStart().isAfterNow()) {
+                    if (ret.getStart().isBeforeNow()) {
```

Comment:
You are starting the `ret` value with the first value in the list. That's why it can be "before now" in all the other cases, it cannot before now, because there is a check in 134. 

I think this can be simplified. Checking "before now" does not make sense. 

Can you try to make it `null` first and handle the `null` in the for loop?